commands/delete-registry.ts

```diff
@@ -0,0 +1,127 @@
+
+import * as vscode from "vscode";
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import { AzureAccountWrapper } from '.././explorer/deploy/azureAccountWrapper';
+import { SubscriptionClient, ResourceManagementClient, SubscriptionModels } from 'azure-arm-resource';
+import { AzureAccount, AzureSession } from '../typings/azure-account.api';
+import { accountProvider } from '../dockerExtension';
+import { RegistryRootNode } from "../explorer/models/registryRootNode";
+import { ServiceClientCredentials } from 'ms-rest';
+import { RegistryNameStatus, RegistryListResult } from "azure-arm-containerregistry/lib/models";
+const teleCmdId: string = 'vscode-docker.deleteRegistry';
+import { AzureCredentialsManager } from '../utils/AzureCredentialsManager';
+import { AzureRegistryNode, AzureLoadingNode, AzureNotSignedInNode } from '../explorer/models/azureRegistryNodes';
+
+/**
+ * delete a registry and all it's associated nested items
+ * @param context : the AzureRegistryNode the user right clicked on to delete
+ */
+export async function deleteRegistry(context?: AzureRegistryNode) {
+    if (!context) {
+        deleteRegistryNoContext();
+        return;
+    }
```

Comment:
Given 2 separate functionalities done in either case you should separate them in methods as in. 
if(!context){
      deleteRegistryNoContext();
}else{
      deleteRegistryContextAvailable();
}